docs/data/scheduler/overview/TimeGridPrimitives.tsx

```diff
@@ -0,0 +1,187 @@
+        agenda: 'work',
+      },
+    ],
+  },
+  {
+    date: createDate(4, 0, 0),
+    events: [
+      {
+        id: '1',
+        start: createDate(4, 7, 30),
+        end: createDate(4, 8, 15),
+        title: 'Footing',
+        agenda: 'personal',
+      },
+    ],
+  },
+  {
+    date: createDate(5, 0, 0),
+    events: [
+      {
+        id: '1',
+        start: createDate(5, 15, 0),
+        end: createDate(5, 15, 45),
+        title: 'Retrospective',
+        agenda: 'work',
+      },
+    ],
+  },
+];
+
+export default function TimeGridPrimitives() {
+  const { scrollableRef, scrollerRef } = useInitialScrollPosition();
+
+  return (
+    <div className={classes.Container}>
+      <TimeGrid.Root className={classes.Root}>
+        <div className={classes.Header}>
+          <div className={classes.TimeAxisHeaderCell} aria-hidden="true" />
+          {days.map((day) => (
+            <div key={day.date.toString()} className={classes.HeaderCell}>
+              {day.date.toFormat('EEE, dd')}
+            </div>
+          ))}
+        </div>
+        <div className={classes.Body} ref={scrollerRef}>
+          <div className={classes.ScrollableContent} ref={scrollableRef} role="row">
+            <div className={classes.TimeAxis} aria-hidden="true">
+              {Array.from({ length: 24 }, (_, hour) => (
+                <div
+                  key={hour}
+                  className={classes.TimeAxisCell}
+                  style={{ '--hour': hour } as React.CSSProperties}
+                >
+                  {hour === 0
+                    ? null
+                    : `${DateTime.now().set({ hour }).toFormat('hh a')}`}
+                </div>
+              ))}
+            </div>
+            {days.map((day) => (
+              <TimeGrid.Column
+                key={day.date.toString()}
+                value={day.date}
+                className={classes.Column}
+              >
+                {day.events.map((event) => (
+                  <TimeGrid.Event
+                    key={event.id}
+                    start={event.start}
+                    end={event.end}
+                    data-agenda={event.agenda}
+                    className={classes.Event}
+                  >
+                    <div className={classes.EventInformation}>
+                      <div className={classes.EventStartTime}>
+                        {event.start.toFormat('hh a')}
+                      </div>
+                      <div className={classes.EventTitle}>{event.title}</div>
+                    </div>
+                  </TimeGrid.Event>
+                ))}
+              </TimeGrid.Column>
+            ))}
+          </div>
+        </div>
+      </TimeGrid.Root>
+    </div>
+  );
+}
+
+function useInitialScrollPosition() {
+  // TODO: Should the automatic scrolling be built-in?
+  const scrollableRef = React.useRef<HTMLDivElement>(null);
+  const scrollerRef = React.useRef<HTMLDivElement>(null);
+
+  useEnhancedEffect(() => {
```

Comment:
Nitpick: WDYT about using the regular `useLayoutEffect` instead of the utils for docs examples? 🤔